src/darwin/Framework/CHIP/MTRSetupPayload.mm

```diff
@@ -206,6 +206,11 @@ @implementation MTRSetupPayload {
     NSNumber * _Nullable _shadowDiscriminator;
 }
 
++ (BOOL)isValidSetupPIN:(NSUInteger)setupPIN
+{
+    return [[MTRSetupPayload alloc] initWithPayload:[NSString stringWithFormat:@"%lu", static_cast<unsigned long>(setupPIN)]] != nil;
```

Comment:
This doesn't do the right thing; what's provided is a passcode, not the whole numeric payload.